doc/collision_detection.md

```diff
@@ -0,0 +1,162 @@
+# Collision detection
```

Comment:
should we add a brief paragraph about how one can do collision detection, ie, you can use forge2d, build your own, use this, etc, with some pros and cons or tips on which to choose?

Reply:
Very good idea!

Reply:
What do you think of the updated text?

Reply:
perfect